caf/base_config/level0/launchpad/iam_role_mapping.tfvars

```diff
@@ -1,47 +1,55 @@
-
 #
 # Services supported: subscriptions, storage accounts and resource groups
 # Can assign roles to: AD groups, AD object ID, AD applications, Managed identities
 #
 role_mapping = {
   built_in_role_mapping = {
-    storage_accounts = {
-      level0 = {
-        "Storage Blob Data Contributor" = {
-          logged_in = {
-            keys = ["user"]
+    subscriptions = {
+      # Required both to create Azure resources on their respective levels
+      # https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/managed_service_identity
+      logged_in_subscription = {
+        "Contributor" = {
+          managed_identities = {
+            keys = ["level1", "level2", "level3", "level4"]
+          }
+        }
+        "User Access Administrator" = {
+          managed_identities = {
+            keys = ["level1", "level2", "level3", "level4"]
           }
         }
       }
+    }
+    # Access storage accounts without an access key
+    storage_accounts = {
       level1 = {
         "Storage Blob Data Contributor" = {
-          logged_in = {
-            keys = ["user"]
+          managed_identities = {
```

Comment:
I think we need to add back the level0 Storage Blob Data Reader role for level1, so the level1 msi can access the launchpad storage.

We also need to to use the previous structure of Data Contributor for the current level and Data Reader for the previous level.

Reply:
Updated